### DISCUSSION TITLE
Can't complete the stripe form fields after redirection

### DISCUSSION THREAD

**Discussion by Anag03** [cypress-io/cypress#28276] [Questions and Help]
Hey guys
On my project, we have a stripe pricing table where the user can click a button to create a subscription and it redirects to this URL
https://checkout.stripe.com/c/pay/cs_test_a1UgwwtQ7POyL3aTstzchdzMxtZl5FvPk0mCfzLeSA0ZP0ToSd95KsZ240#fidkdWxOYHwnPyd1blpxYHZxWjZON1xqRnIyTGFuRDRGU0drb2AwX0NSMTU1fG0xcFYxTG0neCUl

Where cs_test is a unique value for each customer and it's created when the button is clicked. 
The issue is when the new page is opened, cypress commands don't work

I found a web where the same behavior can be replicated https://stripe-tinydemos-pricing-table.glitch.me/
This is the page opened:
![image](https://github.com/cypress-io/cypress/assets/8594828/05c96600-65e7-4006-acae-83c037f13a92)

This is my code 

`it.only('should open the stripe checkout page', function(){
               cy.visit('https://stripe-tinydemos-pricing-table.glitch.me/')
                cy.get('stripe-pricing-table[pricing-table-id="prctbl_1LPkKBAPEf9EarUkgSuVM4vY"]').shadow().find('iframe', {timeout:5000})
                    .its('0.contentDocument.body').should('not.be.undefined')
                    .then(cy.wrap)
                    .find('div[data-testid="price-column"]').eq(1)
                    .find('button').click()
                 cy.origin('https://checkout.stripe.com/', () => {
                    cy.get('input[name="shippingName"]').click().type('Ana');
                });
            });`

cy.origin doesn't work, so I can't complete the fields. 
If you could help me, please

**Comment by portyu9** [comment]
Hi @Anag03 

`cy.origin()` can handle the **top-level navigation** from your application to `checkout.stripe.com`, but it does **not** give Cypress access to nested cross-origin iframes.

For a normal top-level Stripe Checkout element, the pattern is:

```js
cy.origin('https://checkout.stripe.com', () => {
  cy.get('input[name="shippingName"]')
    .should('be.visible')
    .type('Ana')
})
```

Notice that I would use the origin only:

```text
https://checkout.stripe.com
```

rather than relying on a particular Checkout path.

However, if the field you need is rendered inside one of Stripe's secure cross-origin `<iframe>` elements, Cypress cannot automate that frame. `cy.origin()` does not remove the iframe limitation.

That distinction is the first thing I would verify:

1. Confirm the browser has actually navigated to `checkout.stripe.com`.
2. Inside `cy.origin()`, verify that Cypress can query the top-level `body`.
3. Inspect whether the field you want is in the top-level document or inside an `<iframe>`.

If it is inside a Stripe cross-origin iframe, there is no supported Cypress command that makes that iframe directly automatable.

For a robust automated test, I would normally keep the Cypress boundary at your own integration:

- verify your application successfully creates a Stripe Checkout Session,
- verify it redirects to the expected Stripe Checkout URL,
- test your success/cancel return handling,
- test your webhook handling,
- use Stripe's sandbox/test PaymentMethods for server-side payment scenarios.

That tests the part of the flow your application owns without making your suite depend on Stripe's hosted DOM.

If `shippingName` is definitely in the top-level Checkout document rather than an iframe, then the exact Cypress error from `cy.origin()` would be useful, because in that case the iframe limitation alone would not explain the failure.

Please mark this answer as accepted if it helped, thank you.
